### PR TITLE
Sensor parameters

### PR DESCRIPTION
**Pull request by tomaszgieres**

### CODE REVIEW

File: app/assets/javascripts/code/services/bounds_calculator.js [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
+      maxLong.push(stream.max_longitude);
+      minLong.push(stream.min_longitude);
     });
+
+    var north = Math.max.apply(null, maxLat);
+    var south = Math.min.apply(null, minLat);
+    var west = Math.min.apply(null, minLong);
+    var east = Math.max.apply(null, maxLong);
+
+    if(!north){ return; }
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/bounds_calculator.js]: remove `{ }`

File: app/assets/javascripts/code/services/sensors.js [hunk 2]
```diff
@@ -21,12 +33,32 @@ angular.module("aircasting").factory('sensors', ['params', '$http', 'spinner', f
         sensors[sensor.id] = sensor;
       });
       this.sensors = sensors;
+
+      // Parameters
+      var availableParameters = _.uniq(_(this.sensors).map(function(sensor) {
+        return sensor["measurement_type"]
+      }));
+      availableParameters = _.sortBy(availableParameters)
+      availableParameters = _.map(availableParameters, function(availableParameter) {
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/sensors.js]: we could chain this

File: app/assets/javascripts/code/controllers/crowd_map_ctrl.js [hunk 3]
```diff
@@ -124,6 +129,11 @@ function CrowdMapCtrl($scope, $http, params, heat, $window, map, sensors, expand
     $scope.$digest();
   };
 
+  $scope.$watch("sensors.selectedParameter", function(newValue, oldValue) {
+    console.log("watch - selectedParameter()");
+    sensors.onSelectedParameterChange(newValue);
+  }, true)
+
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/crowd_map_ctrl.js]: `$scope.$watch("sensors.selectedParameter", sensors.onSelectedParameterChange, true);`

**Comment by 3v0k4** [comment]
Let's remove some comments too

File: app/assets/javascripts/code/controllers/crowd_map_ctrl.js [hunk 4]
```diff
@@ -26,8 +28,7 @@ function CrowdMapCtrl($scope, $http, params, heat, $window, map, sensors, expand
 
     storage.updateFromDefaults();
 
-    sensors.shouldInitSelected = true;
-    sensors.initSelected();
+    sensors.setDefault();
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/crowd_map_ctrl.js]: rm

File: app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js [hunk 5]
```diff
@@ -47,23 +47,25 @@ function FixedSessionsMapCtrl($scope, params, heat, map, sensors, expandables, s
 
   //fix for json null parsing
   $scope.$watch("params.get('data').sensorId", function(newValue) {
-    console.log("watch - params.get('data').sensorId");
+    console.log("watch - params.get('data').sensorId - ", newValue);
     if(_(newValue).isNull()){
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js]: give a name to the condition

File: app/assets/javascripts/code/aircasting.js [hunk 6]
```diff
@@ -1,15 +1,34 @@
 angular.module('google', []);
 angular.module('aircasting', ['ngRoute', 'ngCookies', "google"], [ "$routeProvider", function($routeProvider){
   var v = $("body").data("version");
+  var promise = fetch('/api/sensors').then(function(r) { return r.json() });
```
Comment (3v0k4) [inline app/assets/javascripts/code/aircasting.js]: sub this $http

File: app/assets/javascripts/code/aircasting.js [hunk 7]
```diff
@@ -1,15 +1,34 @@
...
   $routeProvider.when('/map_crowd',
                       {templateUrl: 'partials/crowd_map.html?v=' + v,
-                        controller: CrowdMapCtrl, reloadOnSearch: false});
+                        controller: CrowdMapCtrl, reloadOnSearch: false,
+                        resolve: {
+                          yellow: function() {
```
Comment (3v0k4) [inline app/assets/javascripts/code/aircasting.js]: extract

File: app/assets/javascripts/code/controllers/crowd_map_ctrl.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 function CrowdMapCtrl($scope, $http, params, heat, $window, map, sensors, expandables, $location, versioner,
-                      storage, storageEvents, infoWindow, rectangles, spinner, functionBlocker, utils, flash, markersClusterer) {
+                      storage, storageEvents, infoWindow, rectangles, spinner, functionBlocker, utils, flash, markersClusterer, yellow) {
+  sensors.setSensors(yellow);
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/crowd_map_ctrl.js]: yellow

File: app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js [hunk 9]
```diff
@@ -47,18 +45,26 @@ function FixedSessionsMapCtrl($scope, params, heat, map, sensors, expandables, s
...
 
   $scope.$watch("sensors.selectedId()", function(newValue, oldValue) {
-    console.log("watch - sensors.selectedId()");
-    if(newValue == oldValue || !newValue){
+    console.log("watch - sensors.selectedId() - ", newValue, " - ", oldValue);
+    if(!newValue){
       return;
     }
+
     params.update({data: {sensorId: newValue}});
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js]: maybe rm

File: app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js [hunk 10]
```diff
@@ -47,18 +45,26 @@ function FixedSessionsMapCtrl($scope, params, heat, map, sensors, expandables, s
...
+
+    // Possibly this is not needed
+    sensors.onSelectedSensorChange(newValue);
```
Comment (3v0k4) [inline app/assets/javascripts/code/controllers/fixed_sessions_map_ctrl.js]: maybe rm

File: app/assets/javascripts/code/services/sensors.js [hunk 11]
```diff
@@ -21,30 +32,38 @@ angular.module("aircasting").factory('sensors', ['params', '$http', 'spinner', f
...
+        });
+      })
+      this.availableParameters = availableParameters;
+
+      // Initialize UI
       this.initSelected();
       spinner.hide();
     },
     initSelected: function() {
+      console.log('initSelected()')
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/sensors.js]: rm `console.log`s

File: app/assets/javascripts/code/aircasting.js (hunk 6 above)
Comment (3v0k4) [reply app/assets/javascripts/code/aircasting.js]:
```
resolve: {
  yellow: ['$http', function($http) {
    return $http.get('/api/sensors', {cache: true})
      .then(function(response){ return response.data})
    }]
}}
```

**Comment by 3v0k4** [comment]
Merged in pieces to staging already

**Comment by tomaszgieres** [comment]
@3v0k4 N I C E !

**Pull request closed without merging**